gossipd/gossipd.c

```diff
@@ -360,6 +360,8 @@ static bool encoding_end_zlib(u8 **encoded, size_t off)
 	/* Successful: copy over and trim */
 	tal_resize(encoded, off + tal_count(z));
 	memcpy(*encoded + off, z, tal_count(z));
+
+	tal_free(z);
```

Comment:
Suspect this is because run-extended-info.c doesn't setup or cleanup tmpctx, so it's NULL.

It should probably do that (hmm, we should initialize tmpctx to 0xdeadbeef or something so it crashes in such a case).  Separate PR though